src/test/java/htsjdk/samtools/cram/build/ContainerFactoryTest.java

```diff
@@ -1,148 +1,297 @@
-                        ReferenceContext.MULTIPLE_REFERENCE_CONTEXT, Slice.NO_ALIGNMENT_START, Slice.NO_ALIGNMENT_SPAN
+                        CRAMStructureTestHelper.createSAMRecordsMapped(RECORDS_PER_SLICE * 3 + 1, 0),
+                        RECORDS_PER_SLICE, 2,
+                        2, Arrays.asList(RECORDS_PER_SLICE * 2, RECORDS_PER_SLICE+ 1),
+                        Arrays.asList(new ReferenceContext(0), new ReferenceContext(0))
                 },
                 {
-                        CRAMStructureTestUtil.getUnplacedRecords(TEST_RECORD_COUNT),
-                        ReferenceContext.UNMAPPED_UNPLACED_CONTEXT, Slice.NO_ALIGNMENT_START, Slice.NO_ALIGNMENT_SPAN
+                        CRAMStructureTestHelper.createSAMRecordsMapped(RECORDS_PER_SLICE * 4, 0),
+                        RECORDS_PER_SLICE, 2,
+                        2, Arrays.asList(RECORDS_PER_SLICE * 2, RECORDS_PER_SLICE * 2),
+                        Arrays.asList(new ReferenceContext(0), new ReferenceContext(0))
                 },
-
-                // these two sets of records are "half" unplaced: they have either a valid reference index or start position,
-                // but not both.  We treat these weird edge cases as unplaced.
-
                 {
-                        CRAMStructureTestUtil.getHalfUnplacedNoRefRecords(TEST_RECORD_COUNT),
-                        ReferenceContext.UNMAPPED_UNPLACED_CONTEXT, Slice.NO_ALIGNMENT_START, Slice.NO_ALIGNMENT_SPAN
+                        // this generates one container because although it has records mapped to two different reference
+                        // contigs, there aren't enough records to reach the minimum single ref threshold
+                        Stream.of(
+                                CRAMStructureTestHelper.createSAMRecordsMapped(RECORDS_PER_SLICE - 1, 0),
+                                CRAMStructureTestHelper.createSAMRecordsMapped(1, 1))
+                                .flatMap(List::stream)
+                                .collect(Collectors.toList()),
+                        RECORDS_PER_SLICE, 1,
+                        1, Arrays.asList(RECORDS_PER_SLICE),
+                        Arrays.asList(ReferenceContext.MULTIPLE_REFERENCE_CONTEXT)
                 },
                 {
-                        CRAMStructureTestUtil.getHalfUnplacedNoStartRecords(TEST_RECORD_COUNT, mappedSequenceId),
-                        ReferenceContext.UNMAPPED_UNPLACED_CONTEXT, Slice.NO_ALIGNMENT_START, Slice.NO_ALIGNMENT_SPAN
+                        // this generates one container because it has some mapped (but not enough to reach the
+                        // minimum single ref threshold), and one unmapped
+                        Stream.of(
+                                CRAMStructureTestHelper.createSAMRecordsMapped(RECORDS_PER_SLICE - 1, 0),
+                                CRAMStructureTestHelper.createSAMRecordsUnmapped(1))
```

Comment:
Do you have any tests with mapped reads mixed with unmapped-but-placed reads?

Reply:
Done.